src/Log/AccessLogDataMap.php

```diff
@@ -94,7 +95,12 @@ public static function createWithStaticResource(
      */
     public function getClientIp() : string
     {
-        return $this->getLocalIp();
+        $headers = ['x-forwarded-for', 'client-ip', 'x-real-ip'];
+        $headerValue = array_reduce($headers, function (?string $default, string $header) {
+            return $this->request->header[$header] ?? $default;
+        });
```

Comment:
I was initially planning to get this info from `request->server`, but I have run the app locally with a simple nginx proxy and it looks swoole does not mimic the behavior of `$_SERVER` 1 to 1.

The headers are there though, so this approach works.

Reply:
Question: should we loop over all of these, or use only the first matched? (Current approach here loops over all, and uses last matched.)

Reply:
Yeah, you are right. I guess I tried too hard to use a more functional approach when a simple foreach loop is probably more convenient here 😅

I'll change it.

Reply:
I get tripped up by that **all**. **the**. **time**. Love functional paradigms, and try to use them... but sometimes they actually are the wrong fit. (Particularly when performance is a consideration!)

Reply:
Same here, haha.

Anyway, I have just pushed the change and used a loop with an early return instead.